src/WinRT.Runtime2/InteropServices/FreeThreadedMarshaler.cs

```diff
@@ -100,14 +100,25 @@ static FreeThreadedMarshaler InitializeInstanceForCurrentThread()
             {
                 // Create the free-threaded marshaler
                 void* marshalerPtr;
-
                 WindowsRuntimeImports.CoCreateFreeThreadedMarshaler(punkOuter: null, ppunkMarshal: &marshalerPtr).Assert();
+                try
+                {
+                    HRESULT hr = IUnknownVftbl.QueryInterfaceUnsafe(marshalerPtr, in WellKnownInterfaceIds.IID_IMarshal, out void* pMarshalObject);
```

Comment:
If you don't actually need the `HRESULT` separately, you can simplify calls like these by always just using `.Assert()` at the end. Same everywhere else you're doing similar calls immediately followed by `Marshal.ThrowExceptionForHR`

Reply:
I meant that unless you need the `HRESULT` explicitly, you can just do this:

```suggestion
                    IUnknownVftbl.QueryInterfaceUnsafe(marshalerPtr, in WellKnownInterfaceIds.IID_IMarshal, out void* pMarshalObject).Assert();
```

Which is more readable 🙂